server/services/alertsService.ts

```diff
@@ -195,6 +203,32 @@ export default class AlertsService {
     }
   }
 
+  private async getPrisonApiAlertSummaryData(prisonApiClient: PrisonApiClient, bookingId: number) {
+    const { activeAlertCount, inactiveAlertCount, alerts } = await prisonApiClient.getInmateDetail(bookingId)
+    const activeAlertTypesFilter: { [key: string]: AlertTypeFilter } = {}
+    const inactiveAlertTypesFilter: { [key: string]: AlertTypeFilter } = {}
+
+    alerts.forEach(alert => {
+      if (alert.active) {
+        activeAlertTypesFilter[alert.alertType] = {
+          code: alert.alertType,
+          description: alert.alertTypeDescription,
+          count: activeAlertTypesFilter[alert.alertType] ? activeAlertTypesFilter[alert.alertType].count + 1 : 1,
+          checked: false,
+        }
+      } else {
+        inactiveAlertTypesFilter[alert.alertType] = {
+          code: alert.alertType,
+          description: alert.alertTypeDescription,
+          count: inactiveAlertTypesFilter[alert.alertType] ? inactiveAlertTypesFilter[alert.alertType].count + 1 : 1,
+          checked: false,
+        }
+      }
+    })
+
+    return { activeAlertCount, inactiveAlertCount, activeAlertTypesFilter, inactiveAlertTypesFilter }
```

Comment:
Think it would be preferable as a reduce to avoid mutation 
Something like....

```javascript
const {active, inactive} = alerts.reduce((output, alert) => {
  if (alert.active) {
    return {
      ...output,
      active: { ...output.active, [alert.alertType]: { ... } }
    }

    return {
      ...output,
      inactive: { ...output.inactive, [alert.alertType]: { ... } }
  }
}, {active: {}, inactive: {}})
```

Reply:
Agreed on trying to avoid the pattern of declaring and then mutating objects if possible, maybe worth a discussion about procedural / functional style at our dev catch up since I don't think we've ever set anything in stone before?

Reply:
Would something like this work:

```

const toAlertTypeFilters = ({ alerts, active }: { alerts: Alert[]; active: boolean }) => {
  return alerts
    .filter(alert => alert.isActive === active)
    .reduce((alertCounts, alert) => {
      // eslint-disable-next-line no-param-reassign
      alertCounts[alert.alertCode.alertTypeCode] = {
        code: alert.alertCode.alertTypeCode,
        description: alert.alertCode.alertTypeDescription,
        count: (alertCounts[alert.alertCode.alertTypeCode] ?? 0) + 1,
        checked: false,
      }
      return alertCounts
    }, {})
}

export const toAlertSummaryData = (alerts: Alert[], alertFlags: AlertFlagLabel[]): AlertSummaryData => {
  return {
    activeAlertCount: alerts.filter(alert => alert.isActive).length,
    activeAlertTypesFilter: toAlertTypeFilters({ alerts, active: true }),
    inactiveAlertCount: alerts.filter(alert => !alert.isActive).length,
    inactiveAlertTypesFilter: toAlertTypeFilters({ alerts, active: false }),
    alertFlags: toAlertFlagLabels(alerts, alertFlags),
  }
}

```

Reply:
Arguably the accumulator is mutating in the reduce though... so is this any better?